pynamodb/connection/base.py

```diff
@@ -780,38 +800,83 @@ def get_exclusive_start_key_map(self, table_name, exclusive_start_key):
             raise TableError("No such table {}".format(table_name))
         return tbl.get_exclusive_start_key_map(exclusive_start_key)
 
-    def delete_item(self,
-                    table_name,
-                    hash_key,
-                    range_key=None,
-                    condition=None,
-                    return_values=None,
-                    return_consumed_capacity=None,
-                    return_item_collection_metrics=None):
-        """
-        Performs the DeleteItem operation and returns the result
-        """
+    def get_operation_kwargs(self,
+                             table_name=None,
+                             hash_key=None,
+                             range_key=None,
+                             key=KEY,
+                             attributes=None,
+                             attributes_to_get=None,
+                             actions=None,
+                             client_request_token=None,
+                             condition=None,
+                             consistent_read=None,
+                             return_values=None,
+                             return_consumed_capacity=None,
+                             return_item_collection_metrics=None,
+                             return_values_on_condition_failure=None):
         self._check_condition('condition', condition)
 
-        operation_kwargs = {TABLE_NAME: table_name}
-        operation_kwargs.update(self.get_identifier_map(table_name, hash_key, range_key))
+        operation_kwargs = {}
         name_placeholders = {}
         expression_attribute_values = {}
 
+        if table_name is not None and hash_key is not None:
```

Comment:
The method name made it seem as though this was used to build kwargs for all operations, so I initially thought that this line would be a bug for operations that don't involve a `hash_key` (such as `DeleteTable`). While this does reduce code, I think this method makes things slightly harder to understand.

Assuming the intent is DRY, consider breaking out chunks of this instead (such as `_get_transaction_kwargs`).

Reply:
yeah, i could have this part be something like `_get_identifiers` and then the rest of it for building kwargs